test/functional/interface_bitcoin_cli.py

```diff
@@ -68,34 +104,30 @@ def run_test(self):
         self.log.info("Test -getinfo returns expected network and blockchain info")
         if self.is_wallet_compiled():
             self.nodes[0].encryptwallet(password)
-        cli_get_info = self.nodes[0].cli('-getinfo').send_cli()
+        cli_get_info_string = self.nodes[0].cli('-getinfo').send_cli()
+        cli_get_info = cli_get_info_string_to_dict(cli_get_info_string)
+
         network_info = self.nodes[0].getnetworkinfo()
         blockchain_info = self.nodes[0].getblockchaininfo()
-        assert_equal(cli_get_info['version'], network_info['version'])
-        assert_equal(cli_get_info['blocks'], blockchain_info['blocks'])
-        assert_equal(cli_get_info['headers'], blockchain_info['headers'])
-        assert_equal(cli_get_info['timeoffset'], network_info['timeoffset'])
-        assert_equal(
-            cli_get_info['connections'],
-            {
-                'in': network_info['connections_in'],
-                'out': network_info['connections_out'],
-                'total': network_info['connections']
-            }
-        )
-        assert_equal(cli_get_info['proxy'], network_info['networks'][0]['proxy'])
-        assert_equal(cli_get_info['difficulty'], blockchain_info['difficulty'])
-        assert_equal(cli_get_info['chain'], blockchain_info['chain'])
+        assert_equal(int(cli_get_info['Version']), network_info['version'])
+        assert_equal(int(cli_get_info['Blocks']), blockchain_info['blocks'])
+        assert_equal(int(cli_get_info['Headers']), blockchain_info['headers'])
+        assert_equal(int(cli_get_info['Time offset (s)']), network_info['timeoffset'])
+        expected_network_info = f"in {network_info['connections_in']}, out {network_info['connections_out']}, total {network_info['connections']}"
+        assert_equal(cli_get_info["Network"], expected_network_info)
+        assert_equal(cli_get_info['Proxy'], network_info['networks'][0]['proxy'])
+        assert_equal(Decimal(cli_get_info['Difficulty']), blockchain_info['difficulty'])
+        assert_equal(cli_get_info['Chain'], blockchain_info['chain'])
 
         if self.is_wallet_compiled():
             self.log.info("Test -getinfo and bitcoin-cli getwalletinfo return expected wallet info")
-            assert_equal(cli_get_info['balance'], BALANCE)
-            assert 'balances' not in cli_get_info.keys()
+            assert_equal(float(cli_get_info['Balance']), BALANCE)
```

Comment:
nit: Comparing of floats in general can lead to comparing errors in general. (Please see https://stackoverflow.com/questions/4548004/how-to-correctly-and-standardly-compare-floats) 

So even though it may not be an issue now and here, it's still a question whether one should consider using `Decimal` comparison or even string comparison.

I don't know exact details here, so feel free to ignore this comment.

Reply:
Yeah I agree, comparing floats directly is not a great idea and we could do something like [this](https://stackoverflow.com/a/33024979/11635160). However I think if we were to use that method a separate PR should be created to replace all float comparisons with it.

Reply:
The others in this file are already using Decimal, no? E.g. `assert_equal(Decimal(cli_get_info['Balance']), amounts[1])`.

Reply:
Yup, they are all using python Decimal for the file. Correct if I am wrong but I think what @kiminuo meant is that using `assert_equal` for floats(`float`/`Decimal`) might lead to comparison errors and maybe another type of assertion should be used to check if the floats are "close enough".

However, I am and very new here, and might not know the nuance behind it.

Reply:
Also changed `float(cli_get_info['Balance'])` to `Decimal(cli_get_info['Balance']` to standardize with the rest of the tests.

Reply:
`Decimal` is not a float

Reply:
My apologies I was confusing myself. Please ignore whatever I said